stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -240,6 +240,7 @@ end
 
 diag(A::Symmetric) = symmetric.(diag(parent(A)), sym_uplo(A.uplo))
 diag(A::Hermitian) = hermitian.(diag(parent(A)), sym_uplo(A.uplo))
+isdiag(A::HermOrSym) = isdiag(parent(A))
```

Comment:
I'm not sure this is correct.  For example:
```jl
julia> A = Symmetric(tril!(rand(3,3)))
3×3 Symmetric{Float64, Matrix{Float64}}:
 0.321068  0.0       0.0
 0.0       0.268771  0.0
 0.0       0.0       0.355687

julia> isdiag(A)
true

julia> isdiag(parent(A))
false
```

Reply:
Yes, I actually thought about wrapping it in the appropriate triangular type, but then we don't have specialized `isdiag` for that. I'll simply remove this line. Thanks for catching it!

Reply:
You could have
```jl
isdiag(::HermOrSym{<:Any,<:Diagonal}) = true
```
I guess

Reply:
I realized we do have specialized code. For AbstractMatrix, we have the following call chain

```julia
isdiag(A) -> isbanded(A, 0, 0) -> istriu(A, 0) && istril(A, 0)
```

and one of the last two checks is basically for free for `*Triangular` (as expected)!

Reply:
This is not very important, but still nice to have:

```julia
julia> using LinearAlgebra, BenchmarkTools

julia> A = Symmetric(diagm(0 => ones(1000)));

julia> @btime isdiag($A)
  723.039 μs (1 allocation: 16 bytes) # vs. 2.860 ms (0 allocations: 0 bytes) on v1.8.4
true
```

The single allocation is due to the triangular wrapper.